app/controllers/users/approvals_controller.rb

```diff
@@ -22,4 +26,15 @@ def reject
     render "users/approvals/approve"
   end
 
+  def check_user
+    unless current_user?(params[:user_id])
+      flash[:notice] = "You are signed in as #{current_user.username}"
+      if params[:origin]
+        developer = Developer.find_by(company_name: params[:company_name])
```

Comment:
Couldn't think of a better way to get the right developer here, attempted to use X-API-KEY header but it's not present in this request and couldn't find a way to make it be.
Open to suggestions or just leave as is.